internal/controller/processor.go

```diff
@@ -76,6 +76,7 @@ func (processor *AppConfigurationProviderProcessor) processFullReconciliation()
 	processor.RefreshOptions.ConfigMapSettingPopulated = true
 	processor.RefreshOptions.updatedKeyValueETags = updatedSettings.KeyValueETags
 	processor.RefreshOptions.updatedFeatureFlagETags = updatedSettings.FeatureFlagETags
+	processor.RefreshOptions.updatedSentinelETags = updatedSettings.SentinelETags
```

Comment:
Shall we check if the returned `updatedSettings.SentinelETags` is not nil, then we also change the `processor.RefreshOptions.sentinelChanged=true`? Then we don't have to add additional check in `Finish()`

Reply:
And we can also ensure that the `processor.ReconciliationState.SentinelETags` would never be set to nil

Reply:
There may be 2 scenarios when shouldReconcile is true. User adds monitoring section or removes it. If no sentinel is setup, we need to make sure the `ReconciliationState.SentinelETags` be updated to empty list.

Reply:
I agree that sentinelETags should not be nil.